pygmyui/pygmy/views.py

```diff
@@ -61,8 +61,12 @@ def link_shortener(request):
                               context=API_ERROR(e.args[0]),
                               status=401)
             except (ObjectNotFound, InvalidInput) as e:
+                err = {}
+                err["error"] = (" ".join(API_ERROR(e.args[0])['error'])
+                            if isinstance(API_ERROR(e.args[0])['error'],list) 
+                            else API_ERROR(e.args[0])['error'])
```

Comment:
Good find. This PR will fix the issue but I reckon this same issue might occur for some other endpoints also and that would require us to copy this code there. I think we should fix the source of the problem. This https://github.com/amitt001/pygmy/blob/master/pygmy/rest/shorturl.py#L44 is returning a dict `{'error': ['URL is already a pygmy shortened link.'], 'long_url': ['URL is already a pygmy shortened link.']}` with a list as the value. This should be a string. Would you mind updating the PR to fix this?

Reply:
I have changed code, and convert error message to string in API part. But a question came to my mind, isn't it possible to have multiple error in the same time? If it possible, is converting to string good idea?

Reply:
Thanks!
yes, it's possible to have multiple errors at the same time but not with how this webapp works. All form validation is already done on the Django app side and this is the only check that's performed in the backed.
But you are right that it may not be the best idea to join all errors if API is used independently of the webapp. One way, I think, this issue can be solved is by keep on passing error list and on webapp side(frontend or Django app) format it in the form that it shows all the errors properly. 
Frontend already supports this for form validation

<img width="1061" alt="Screenshot 2019-06-21 at 14 50 22" src="https://user-images.githubusercontent.com/7390944/59923640-303c9e00-9434-11e9-90b7-4ea0227a3c21.png">